hail/src/main/scala/is/hail/expr/ir/functions/Functions.scala

```diff
@@ -251,34 +251,34 @@ abstract class RegistryFunctions {
         srvb.offset)
   }
 
-  def registerCode(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[Code[_]]) => Code[_]) {
+  def registerCode(mname: String, aTypes: Array[Type], rType: Type)(impl: (EmitRegion, Array[(PType, Code[_])]) => Code[_]) {
     IRFunctionRegistry.addIRFunction(new IRFunctionWithoutMissingness {
       override val name: String = mname
 
       override val argTypes: Seq[Type] = aTypes
 
       override val returnType: Type = rType
 
-      override def apply(r: EmitRegion, args: Code[_]*): Code[_] = impl(r, args.toArray)
+      override def apply(r: EmitRegion, args: (PType, Code[_])*): Code[_] = impl(r, args.toArray)
```

Comment:
Why not just have args be Array[(Type, Code[_])], since only use is impl(r, args.toArray)

Reply:
Ah, never mind, this allows us to call IRFunctionWithoutMissingness(r, (PType, Code[_]), (PType, Code[_]) ... varargs, that get shown as a Seq, but passing a Seq[Code[_]] isn't the same...

<img width="215" alt="Screen Shot 2019-07-03 at 1 05 27 PM" src="https://user-images.githubusercontent.com/5543229/60610997-3cf5b600-9d93-11e9-8fd1-43af8d7036b1.png">
<img width="689" alt="Screen Shot 2019-07-03 at 1 06 00 PM" src="https://user-images.githubusercontent.com/5543229/60611023-50088600-9d93-11e9-9b56-928ec45c14d1.png">

Reply:
yeah, that's a good change, but I don't really want to introduce more semantic changes in this PR beyond what's necessary.